### PR TITLE
Tidy store operations

### PR DESCRIPTION
**Pull request by dnahodil**
This tidies some methods which were in the wrong place or which seemed to perform an operation an Ext Store should already be able to do.

### CODE REVIEW

File: web-app/js/portal/cart/DownloadPanelItemTemplate.js [hunk 1]
```diff
@@ -176,7 +176,7 @@ Portal.cart.DownloadPanelItemTemplate = Ext.extend(Ext.XTemplate, {
 
     _removeButtonOnClick: function(button) {
         var collectionId = this.getIdFromButtonContainerId(button, "removeButtonId");
-        var record = this.dataCollectionStore.getRecordFromUuid(collectionId);
+        var record = this.dataCollectionStore.dataCollectionWithUuid(collectionId);
```
Comment (jkburges) [inline web-app/js/portal/cart/DownloadPanelItemTemplate.js]: `getDataCollectionWithUuid`?

Comment (dnahodil) [inline web-app/js/portal/cart/DownloadPanelItemTemplate.js]: Yeah. Will change (I'll change the metadata one, too)

Comment (dnahodil) [inline web-app/js/portal/cart/DownloadPanelItemTemplate.js]: Given the DataCollectionStore only holds DataCollections, and there are other get methods already (`getAt(...)`, `getById(...)`) could I simply call it `getByUuid(...)`?

Comment (jkburges) [inline web-app/js/portal/cart/DownloadPanelItemTemplate.js]: > could I simply call it getByUuid(...)?

Yes, this is better.

**Comment by dnahodil** [comment]
Changes made

File: src/test/javascript/portal/search/FacetedSearchResultsDataViewSpec.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ describe("Portal.search.FacetedSearchResultsDataView", function() {
     beforeEach(function() {
         facetedSearchDataView = new Portal.search.FacetedSearchResultsDataView({
             dataCollectionStore: {
-                isRecordActive: noOp,
+                getWithUuid: noOp,
```
Comment (jkburges) [inline src/test/javascript/portal/search/FacetedSearchResultsDataViewSpec.js]: I thought you were going with `getByUuid`?  "with" doesn't really make sense with out the "dataCollection" bit before it.

Comment (jkburges) [inline src/test/javascript/portal/search/FacetedSearchResultsDataViewSpec.js]: Keep trying, you'll get there.

Comment (dnahodil) [inline src/test/javascript/portal/search/FacetedSearchResultsDataViewSpec.js]: Whoops

**Comment by dnahodil** [comment]
Sit tight. I've found a bug in my changes.

**Comment by dnahodil** [comment]
GTG

**Comment by jkburges** [comment]
image

**Comment by dnahodil** [comment]
Beautiful :cry:

**Pull request merged**